src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -72,25 +72,20 @@ partitions_request_iterator validate_partitions(
   reassignable_topic_response topic_response,
   std::vector<model::node_id> all_node_ids) {
     // An undefined replicas vector is not an error, see "Replicas" in the
-    // AlterPartitionReassignmentsRequest schemata. Instead of checking for
-    // a null replicas vector in every call to validate_replicas, simply put
-    // those elements first and use iterator magic to run validation on
-    // defined replicas only.
-    auto valid_partitions_end = std::partition(
-      begin, end, [](const reassignable_partition& partition) {
-          return !partition.replicas.has_value();
-      });
```

Comment:
From live review:
This change may not be necessary anymore. Changing the ordering should not result in skipped elements.

Reply:
Validation is run on all replica sets, even the undefined ones. So reordering here with std::partition does not accomplish anything useful. I'll update the commit message to reflect this.